JASP-Engine/JASPgraphs/R/parseThis.R

```diff
@@ -0,0 +1,40 @@
+#' @title Tell that this string should be parsed.
+#'
+#' @param x the object to be parsed, or the object to test if it will be parsed.
+#'
+#' @details \code{parseThis} adds an attribute to a character or factor such that JASPgraphs knows it should be
+#' parsed. \code{isParsed} return \code{TRUE} if a character has said attribute.
+
+
+#' @rdname parseThis
+#' @export
+parseThis <- function(x) {
+  UseMethod("parseThis", x)
+}
+
+#' @export
+parseThis.character <- function(x) {attr(x, "parse") <- TRUE; x}
+#' @export
+parseThis.factor    <- function(x) {attr(x, "parse") <- TRUE; x}
```

Comment:
Do you really need parseThis? It only seems to do something with types it doesn't actually mind.
E.g., for characters and factors it adds "parse" but then doesn't need to parse them, however for data.frames it doesn't add "parse" and then it does need to parse them.
Maybe I'm overlooking something